File: Documentation/API/Parsers/Midi/ReadTimeSignatureChangesFromMidiData.cpp
#include <iostream>

#include "RhythmGameUtilities/File.hpp"
#include "RhythmGameUtilities/Parsers/Midi.hpp"

using namespace RhythmGameUtilities;

auto main() -> int
{
    auto bytes = ReadBytesFromFile("./tests/Mocks/song.mid");

    auto timeSignatureChanges = ReadTimeSignatureChangesFromMidiData(bytes);

    std::cout << size(timeSignatureChanges) << std::endl; // 4

    return 0;
}

File: Documentation/API/Parsers/Midi/ReadTimeSignatureChangesFromMidiData.cs
using System;
using System.IO;
using RhythmGameUtilities;

var bytes = File.ReadAllBytes("./Tests/mocks/song.mid");

var timeSignatureChanges = Midi.ReadTimeSignatureChangesFromMidiData(bytes);

Console.WriteLine(timeSignatureChanges.Length); // 1
